app/screens/integration_selector/integration_selector.tsx

```diff
@@ -288,10 +277,25 @@ function IntegrationSelector(
         }
     }, [options, getDynamicOptions, integrationData]);
 
+    const handleSelectProfile = useCallback((user: UserProfile): void => {
+        if (!isMultiselect) {
+            handleSelect(user);
+            close();
+        }
+
+        setSelectedIds((current) => handleIdSelection(dataSource, current, user));
+    }, [isMultiselect, handleIdSelection, handleSelect, close, dataSource]);
+
     const onHandleMultiselectSubmit = useCallback(() => {
-        handleSelect(Object.values(multiselectSelected));
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            // New multiselect
+            handleSelect(Object.values(selectedIds) as UserProfile[]);
+        } else {
+            // Legacy multiselect
+            handleSelect(Object.values(multiselectSelected));
+        }
         close();
-    }, [multiselectSelected, handleSelect]);
+    }, [multiselectSelected, selectedIds, handleSelect]);
```

Comment:
Why not use the `multiselectSelected` for users like we do for channels and options? We can always form the selectedIds just by getting `Object.keys(multiselectSelected)`, right?

Reply:
I'd like to remove it in the future so all of them use selectedIds and remove multiselectSelected from the code.

Reply:
I'll change it in the options pr 👌, I have the channel list pr ready too ☺️